### PR TITLE
PERF: increase the minimum number of elements to use numexpr for ops from 1e4 to 1e6

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
See https://github.com/pandas-dev/pandas/issues/40500 for the analysis.

### CODE REVIEW

File: pandas/core/computation/expressions.py [hunk 1]
```diff
@@ -40,7 +40,7 @@
 }
 
 # the minimum prod shape that we will use numexpr
-_MIN_ELEMENTS = 10000
+_MIN_ELEMENTS = 1000000
```
Comment (simonjayhawkins) [inline pandas/core/computation/expressions.py:43]:
```suggestion
_MIN_ELEMENTS = 1_000_000
```

**Comment by jreback** [comment]
prob need to update the docs a bit here https://pandas.pydata.org/docs/user_guide/enhancingperf.html

**Comment by jorisvandenbossche** [comment]
> prob need to update the docs a bit here https://pandas.pydata.org/docs/user_guide/enhancingperf.html

That page is about using numexpr in `eval`/`query`, so that is independent from this change. On the basic page is a section about accelerated ops: https://pandas.pydata.org/docs/user_guide/basics.html?highlight=numexpr#accelerated-operations 
But it doesn't mention the minimum number of items, so that doesn't need to be updated (we should update the table id at some point switching to the ArrayManager, as that will be influenced, but so that's still for the future)

**Review by jbrockmendel — APPROVED** [review]
LGTM pending green

**Comment by jbrockmendel** [comment]
Looks like this wasnt green when merged, could be responsible for broken CI?

**Pull request merged**